core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -203,7 +204,22 @@ func (i *pluginOracleCreator) Create(donID uint32, config cctypes.OCR3ConfigWith
 	if err != nil {
 		return nil, err
 	}
-	return oracle, nil
+
+	closeFunc := func() error {
```

Comment:
Wouldn't it be simpler to just pass all the closeable objects as slice and let the `wrappedOracle` do its job? This lambda function is a bit hacky to me. I mean something like this.

```go
closeableResources := make([]Closeable, 0)
closeableResources = append(closeableResources, maps.Values(chainReaders))
closeableResources = append(closeableResources, maps.Values(chainWriters))

return newWrappedOracle(oracle, closeableResource), nil
```

And then iterate through that slice in the wrapped oracle? This way you can pass there anything that implements `Close()`

Reply:
Yeah this better than my suggestion, but you should still consider using `services.CloseAll` internally.